### PR TITLE
Add sharding config file as default value for --sharding_config_path

### PR DESCRIPTION
**Pull request by samanvp**
We also drop `--partition_config_path` flag. This is breaking change, however, considering that we completely modified the format of sharding config file, this break is not avoidable.

### CODE REVIEW

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 1]
```diff
@@ -199,39 +211,62 @@ def add_arguments(self, parser):
...
-        raise ValueError('--update_schema_on_append requires --append to be '
-                         'true.')
+    if parsed_args.update_schema_on_append and not parsed_args.append:
+      raise ValueError('--update_schema_on_append requires --append to be '
+                       'true.')
+    if (not parsed_args.sharding_config_path or
+        not parsed_args.sharding_config_path.strip()):
+      raise ValueError(
+          '--sharding_config_path must point to a valid config file.')
+    # Ensuring (not) existence of output tables is aligned with --append value.
```
Comment (tneymanov) [inline gcp_variant_transforms/options/variant_transform_options.py]: I don't think you are checking that _sample table has already been created here. We need to add that as well.

File: gcp_variant_transforms/libs/bigquery_util.py [hunk 2]
```diff
@@ -151,23 +151,19 @@ def raise_error_if_dataset_not_exists(client, project_id, dataset_id):
       raise
 
 
-def raise_error_if_table_exists(client, project_id, dataset_id, table_id):
-  # type: (bigquery.BigqueryV2, str, str, str) -> None
+def does_table_exist(client, project_id, dataset_id, table_id):
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/libs/bigquery_util.py]: Done.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 3]
```diff
@@ -199,39 +211,62 @@ def add_arguments(self, parser):
               'For instance, [0, None, 1] will become '
               '[0, `null_numeric_value_replacement`, 1].'))
 
+
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: done.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 4]
```diff
@@ -199,39 +211,62 @@ def add_arguments(self, parser):
...
-                                                project_id,
-                                                dataset_id,
-                                                table_id)
+        all_output_tables.append(
+            sample_info_table_schema_generator.compose_table_name(
+                table_id, sample_info_table_schema_generator.TABLE_SUFFIX))
+      sharding = variant_sharding.VariantSharding(
+          parsed_args.sharding_config_path)
+      num_shards = sharding.get_num_shards()
+      # In case there is no residual in config we will ignore the last shahrd.
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: done.

File: gcp_variant_transforms/options/variant_transform_options.py (hunk 1 above)
Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: A few line below we add it to `all_output_tables` and we verify it along with all other output tables.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 5]
```diff
@@ -136,6 +136,18 @@ def add_arguments(self, parser):
...
                         help='BigQuery table to store the results.')
+    parser.add_argument(
+        '--sharding_config_path',
+        default=('gcp_variant_transforms/data/sharding_configs/'
+                 'homo_sapiens_default.yaml'),
+        help=('File containing list of output tables, their name suffixes, and '
+              'approximate number of total base pairs which is used to conduct '
+              'BigQuery integer range partitioning. Default value is set to a '
+              'file which is optimized for the human genome. It results in one '
+              'table per chromosome (overall 25 BigQuery tables). You can '
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: done.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 6]
```diff
@@ -199,39 +211,62 @@ def add_arguments(self, parser):
...
+                       'true.')
+    if (not parsed_args.sharding_config_path or
+        not parsed_args.sharding_config_path.strip()):
+      raise ValueError(
+          '--sharding_config_path must point to a valid config file.')
+    # Ensuring (not) existence of output tables is aligned with --append value.
+    if parsed_args.output_table:
+      if not client:
+        credentials = GoogleCredentials.get_application_default().create_scoped(
+            ['https://www.googleapis.com/auth/bigquery'])
```
[the remark this answers is not in the document]
Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: This is a valid point, but I rather not to do it in this PR. Let me do it in another PR.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 7]
```diff
@@ -136,6 +136,17 @@ def add_arguments(self, parser):
...
+        '--sharding_config_path',
+        default=('gcp_variant_transforms/data/sharding_configs/'
+                 'homo_sapiens_default.yaml'),
+        help=('File containing list of output tables, their name suffixes, and '
+              'approximate number of total base pairs which is used to conduct '
+              'BigQuery integer range partitioning. Default value is set to a '
+              'file which is optimized for the human genome. It results in one '
+              'table per chromosome (overall 25 BigQuery tables). For more '
+              'information visit gcp_variant_transforms/data/sharding_configs/'
+              'homo_sapiens_default.yaml'))
```
Comment (tneymanov) [inline gcp_variant_transforms/options/variant_transform_options.py]: Hmm, I actually meant to a proper doc - gcp-variant-trannsforms/docs/sharding.md. It's outdated right now, but we are going to modify it at some point to reflect the new schema, I presume.

Comment (samanvp) [reply gcp_variant_transforms/options/variant_transform_options.py]: Sorry, done.

**Pull request merged**